modules/staking_pool/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{decl_error, decl_event, decl_module, decl_storage, ensure, traits::Get};
+use orml_traits::BasicCurrency;
+use rstd::prelude::*;
+use sp_runtime::{
+	traits::{AccountIdConversion, Saturating, UniqueSaturatedInto, Zero},
+	DispatchResult, ModuleId,
+};
+use support::{
+	EraIndex, ExchangeRate, NomineesProvider, OnCommission, OnNewEra, PolkadotBridge, PolkadotBridgeCall,
+	PolkadotBridgeState, PolkadotBridgeType, Rate, Ratio,
+};
+use system::{self as system, ensure_root};
+
+const MODULE_ID: ModuleId = ModuleId(*b"aca/stkp");
+
+type StakingBalanceOf<T> = <<T as Trait>::StakingCurrency as BasicCurrency<<T as system::Trait>::AccountId>>::Balance;
+type LiquidBalanceOf<T> = <<T as Trait>::LiquidCurrency as BasicCurrency<<T as system::Trait>::AccountId>>::Balance;
+type PolkadotAccountIdOf<T> =
+	<<T as Trait>::Bridge as PolkadotBridgeType<<T as system::Trait>::BlockNumber>>::PolkadotAccountId;
+
+pub trait Trait: system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+	type StakingCurrency: BasicCurrency<Self::AccountId>;
+	type LiquidCurrency: BasicCurrency<Self::AccountId>;
+	type Nominees: NomineesProvider<PolkadotAccountIdOf<Self>>;
+	type OnCommission: OnCommission<LiquidBalanceOf<Self>>;
+	type Bridge: PolkadotBridge<Self::BlockNumber, StakingBalanceOf<Self>, Self::AccountId>;
+	type MaxBondRatio: Get<Ratio>;
+	type MinBondRatio: Get<Ratio>;
+	type MaxClaimFee: Get<Rate>;
+}
+
+decl_event!(
+	pub enum Event<T>
+	where
+		<T as system::Trait>::AccountId,
+		StakingBalance = StakingBalanceOf<T>,
+		LiquidBalance = LiquidBalanceOf<T>,
+	{
+		Mint(AccountId, StakingBalance, LiquidBalance),
+	}
+);
+
+decl_error! {
+	/// Error for staking pool module.
+	pub enum Error for Module<T: Trait> {
+		DOTNotEnough,
```

Comment:
avoid `DOT` in code directly, because we will support KSM initially, and later maybe other chain's token